src/backend/distributed/shared_library_init.c

```diff
@@ -886,6 +886,18 @@ RegisterCitusConfigVariables(void)
 		GUC_STANDARD,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
```

Comment:
Ok, I'm convinced that this is a useful GUC for any user. In the end, if you have circular dependency views for local-only tables, we are NOT allowing that. But, as long as you decide to distribute the underlying tables, we should have a way to allow local circular dependencies.

So, lets not make this `GUC_SUPERUSER_ONLY | GUC_NO_SHOW_ALL` but just a `GUC_STANDARD` such that users can also change this.


And, we are already NOT distributing some of the local views already, as the comment also indicates (and also pg_dist_object lacks this view):
```SQL
CREATE OR REPLACE VIEW v_test AS SELECT * FROM pg_local;
WARNING:  "view v_test" has dependency to "table pg_local" that is not in Citus' metadata
DETAIL:  "view v_test" will be created only locally
HINT:  Distribute "table pg_local" first to distribute "view v_test"
CREATE VIEW
```

So, we should find a better name for the GUC. What do you think about `citus.enable_local_view_dependency_checks`

In fact, when the GUC is off, it'd be awesome to skip these WARNINGs. Can we do that?
```SQL
WARNING:  "view v_test" has dependency to "table pg_local" that is not in Citus' metadata
DETAIL:  "view v_test" will be created only locally
HINT:  Distribute "table pg_local" first to distribute "view v_test"
```

@velioglu any thoughts?

Reply:
Talking with @velioglu, we might consider the name to be more generic, like `enable_local_dependency_checks` for now, and if needed, we can add other object types to follow the same GUC.